pkg/virt-launcher/virtwrap/converter/archvoncerter_test.go

```diff
@@ -0,0 +1,40 @@
+/* Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright the KubeVirt Authors.
+ *
+ */
+
+package converter
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("archConverter", func() {
```

Comment:
I would expect here some tests for `addGraphicsDevice` and `isUSBNeeded` too

Reply:
Those are already covered for each arch in `converter_test.go` by comparing them to pre-defined expected xml definitions. As such i don't think it makes much sense to duplicate this code again.
After all the code inside those functions did not change, we just now call it a bit differently. As such it is covered by existing tests.

Reply:
A bit similar to https://github.com/kubevirt/kubevirt/pull/11907#discussion_r1611613115, the problem as I see it is that code can change. You are right that today `converter_test.go` might be testing it since this PR only brings refactoring and not an actual change of logic, however who knows how the code will look like tomorrow. For example, the arch converter interface can be used in other areas in the future, or the converter logic might change.

The whole concept behind unit tests is that every unit tests itself in a cohesive way. Converter has different tasks, as it converts the vmi into a libvirt XML. While it's true that it uses the arch converter, it uses it as a block box and assumes it's already working. It is not its job to test it and validate it.

One last point is that the most valuable merit of tests in general is NOT to ensure the code is working as expected (this can be validated manually as well, sometimes more easily) but to allow easy refactoring in the future and hence ease future development. Building upon an implicit assumption that the code is being tested by a different unit in an implicit way goes against this principle IMO.

Reply:
I see your point, i think the difference here is what would you think of as a unit? For me in golang that has always been a package, as that is the smallest piece in golang that you can import. As such what a package should be testing is that the exported functionality is stable.
In this case that would be `converter`, as it is the exposed type. On the other hand `archConverter` exists only inside the package, it's only use is to be used by converter as a convenient tool to outsource arch-specific code. So my argument is that if `converter_test.go` does not cover it anymore, it is also no longer needed in the `converter` and should just be removed.

Reply:
That's a fair argument.
I tend to see a unit as the smaller form of an "entity" that provides some kind of a service. The reason it is valuable to look at is this way is that even inside the package, this interface can end up being used for something completely different. If we don't see the interface as a unit, then it remains untested in this case.

But in any case, I don't think it should be blocking for this PR.